vochain/indexer/process.go

```diff
@@ -312,3 +318,29 @@ func (idx *Indexer) updateProcess(ctx context.Context, queries *indexerdb.Querie
 	}
 	return nil
 }
+
+// updateProcessMetadata synchronize title and description
+// with the information fetched over IPFS.
+func (idx *Indexer) UpdateProcessMetadata(d data.Storage, pid []byte, uri string) error {
+	// Try to retrieve the election metadata
+	if d != nil {
+		stgCtx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		// TODO: abstract all of this somewhere else and just call from here
+		metadataBytes, err := d.Retrieve(stgCtx, uri, MaxOffchainFileSize)
+		if err != nil {
+			log.Warnf("cannot get metadata from %s: %v", election.MetadataURL, err)
+		} else {
+			// if metadata exists and is not encrypted, add it to the indexed election
+			// if the metadata is not encrypted, unmarshal it, otherwise store it as bytes
+			if !election.ElectionMode.EncryptedMetaData {
+				electionMetadata := api.ElectionMetadata{}
+				if err := json.Unmarshal(metadataBytes, &electionMetadata); err != nil {
+					log.Warnf("cannot unmarshal metadata from %s: %v", election.MetadataURL, err)
+				}
+				election.Metadata = &electionMetadata
```

Comment:
this doesn't work at all, doesn't even compile, it's just a copypaste from api/elections.go